tsdb/index/index.go

```diff
@@ -51,6 +51,9 @@ const (
 	indexFilename = "index"
 
 	seriesByteAlign = 16
+
+	// checkContextEveryNIterations is used in some tight loops to check if the context is done.
+	checkContextEveryNIterations = 100
```

Comment:
I think this should be `1024`
- I feel like `100` is too often. How often do we want to check? I'd say every 10ms or even 100ms would be enough. For 10ms, this would mean that each check takes 100 microseconds, and that sounds like a lot (do we have numbers?)
- Can we make it a power of two like `1024`? We don't really care if it's every 1000 or every 1024 (or 100 vs 128), but it's way cheaper for the CPU to do the latter operation. (I did the benchmarks, and there's a noticeable difference when switching to 128. Interestingly, same difference as checking `count == count>>7<<7` but for some reason, checking `count & (1<<7-1) == 0` is less performant)

Reply:
I'm also happy to follow up later with a separate PR and benchmarks that would support this comment.

Reply:
Agreed that checking every 100ms is enough.
We didn't have timings; I guessed that 100 was right.
The case that prompted this PR had a 7KB regexp with a 40-deep stack.
Also the CPU was throttled because it was well over its request.

Reply:
I don't have a favorite in this race, we've seen 1000 be effective in a test we did.  @bboreham are you ok changing to 1024?

Reply:
Given that we put out the fire in our own infra, I would like to see one timing of one comparable case before changing this.

Reply:
Let's merge with 100 and I'll bring proper benchmarks to update the value later.

Reply:
Follow up: https://github.com/prometheus/prometheus/pull/14118#issue-2303036892